src/spikeinterface/widgets/motion.py

```diff
@@ -96,55 +94,241 @@ def plot_matplotlib(self, data_plot, **backend_kwargs):
+            assert sorting_analyzer.has_extension(
+                "spike_locations"
+            ), "The sorting analyzer must have the 'spike_locations' extension to use this function"
+            peak_locations = sorting_analyzer.get_extension("spike_locations").get_data()
+            if color_amplitude:
+                assert sorting_analyzer.has_extension("spike_amplitudes"), (
+                    "The sorting analyzer must have the 'spike_amplitudes' extension to use color_amplitude=True. "
+                    "You can compute it or set color_amplitude=False."
+                )
+            if sorting_analyzer.has_extension("spike_amplitudes"):
+                peak_amplitudes = sorting_analyzer.get_extension("spike_amplitudes").get_data()
+            else:
+                peak_amplitudes = None
+        times = recording.get_times(segment_index=segment_index) if recording is not None else None
+
+        if segment_index is None:
+            assert (
+                len(np.unique(peaks["segment_index"])) == 1
+            ), "segment_index must be specified if there are multiple segments"
+            segment_index = 0
+        else:
+            peak_mask = peaks["segment_index"] == segment_index
+            peaks = peaks[peak_mask]
+            peak_locations = peak_locations[peak_mask]
+            if peak_amplitudes is not None:
+                peak_amplitudes = peak_amplitudes[peak_mask]
+
+        if recording is not None:
+            sampling_frequency = recording.sampling_frequency
+            times = recording.get_times(segment_index=segment_index)
+        else:
+            times = None
+
+        plot_data = dict(
+            peaks=peaks,
+            peak_locations=peak_locations,
+            peak_amplitudes=peak_amplitudes,
+            direction=direction,
+            times=times,
+            sampling_frequency=sampling_frequency,
+            segment_index=segment_index,
+            depth_lim=depth_lim,
+            color_amplitude=color_amplitude,
+            color=color,
+            scatter_decimate=scatter_decimate,
+            cmap=cmap,
+            clim=clim,
+            alpha=alpha,
+            recording=recording,
+        )
+        BaseWidget.__init__(self, plot_data, backend=backend, **backend_kwargs)
+
+    def plot_matplotlib(self, data_plot, **backend_kwargs):
+        import matplotlib.pyplot as plt
+        from matplotlib.colors import Normalize
+        from .utils_matplotlib import make_mpl_figure
+
+        from spikeinterface.sortingcomponents.motion_interpolation import correct_motion_on_peaks
+
+        dp = to_attr(data_plot)
+
+        assert backend_kwargs["axes"] is None, "axes argument is not allowed in DriftRasterMapWidget. Use ax instead."
+
+        self.figure, self.axes, self.ax = make_mpl_figure(**backend_kwargs)
+
+        if dp.times is None:
+            peak_times = dp.peaks["sample_index"] / dp.sampling_frequency
```

Comment:
Sorry completely forgot, is this a place to use `time_from_sample_index` etc. to handle `t_start` and `time vector`?

Reply:
The problem here is that recording can be None (in that case sampling_frequency is required)